superset/security/manager.py

```diff
@@ -668,13 +676,16 @@ def create_missing_perms(self) -> None:
             if pv.permission and pv.view_menu:
                 all_pvs.add((pv.permission.name, pv.view_menu.name))
 
-        def merge_pv(view_menu: str, perm: str) -> None:
+        def merge_pv(view_menu: str, perm: Optional[str]) -> None:
             """Create permission view menu only if it doesn't exist"""
             if view_menu and perm and (view_menu, perm) not in all_pvs:
                 self.add_permission_view_menu(view_menu, perm)
 
         logger.info("Creating missing datasource permissions.")
-        datasources = ConnectorRegistry.get_all_datasources(self.get_session)
+        # pylint: disable=import-outside-toplevel
+        from superset.datasource.dao import DatasourceDAO
+
+        datasources = DatasourceDAO.get_all_sqlatables_datasources(self.get_session)
```

Comment:
I was noticing that SqlaTable has some class methods that access security manager, thus the circular dependencies. I think for the next PR we could just move all of this logic into the SqlaTable class, especially because there's only one model now that Native Druid is removed.